### PR TITLE
Add rust 1.66 container

### PR DESCRIPTION
**Pull request by jlausuch**
https://progress.opensuse.org/issues/122956

### CODE REVIEW

**Review by dcermak — CHANGES_REQUESTED** [review]
This is not sufficient, you need to add the rust container to https://github.com/SUSE/BCI-tests/blob/bb59492ee6f7616245f37c07016c0313ac43d13a/bci_tester/data.py#L351 as well.

**Comment by jlausuch** [comment]
> This is not sufficient, you need to add the rust container to
> 
> https://github.com/SUSE/BCI-tests/blob/bb59492ee6f7616245f37c07016c0313ac43d13a/bci_tester/data.py#L351
> as well.

Oops, I forgot. Thanks and updated :)

File: bci_tester/data.py [hunk 1]
```diff
@@ -354,7 +354,7 @@ def create_BCI(
         image_type="dockerfile",
         available_versions=["15.4"],
     )
-    for rust_version in ("1.64", "1.65")
+    for rust_version in ("1.64", "1.65", "1.66")
 ]
 (RUST_1_64_CONTAINER, RUST_1_65_CONTAINER) = RUST_CONTAINERS
```
Comment (dcermak) [inline bci_tester/data.py:359]:
```suggestion
(RUST_1_64_CONTAINER, RUST_1_65_CONTAINER, RUST_1_66_CONTAINER) = RUST_CONTAINERS
```

And you need to add RUST_1_66_CONTAINER further below as well (and maybe to the metadata tests as well).

Comment (jlausuch) [reply bci_tester/data.py]: Amended. metadata uses the anchor `RUST_CONTAINERS`

File: bci_tester/data.py [hunk 2]
```diff
@@ -489,6 +489,7 @@ def create_BCI(
     RUBY_25_CONTAINER,
     RUST_1_64_CONTAINER,
     RUST_1_65_CONTAINER,
+    RUST_1_66_CONTAINER
```
Comment (dcermak) [inline bci_tester/data.py:492]:
```suggestion
    RUST_1_66_CONTAINER,
```

**Comment by dcermak** [comment]
And please change the commit message as well, it's no longer just adding the marker.

**Review by dcermak — APPROVED** [review]
LGTM, thanks!

**Comment by jlausuch** [comment]
The failing pipeline says
```
AssertionError: Unexpected exit code 1 for CommandResult(command=b'docker pull registry.suse.com/bci/rust:1.66', exit_status=1, stdout=None, stderr=b'Error response from daemon: manifest for registry.suse.com/bci/rust:1.66 not found: manifest unknown: manifest unknown\n')
```
But this is expected, as the container is not tested yet, therefore not released. Shall we merge anyway?

**Comment by dcermak** [comment]
> But this is expected, as the container is not tested yet, therefore not released. Shall we merge anyway?

It will also make your openQA tests fail. If you are ok with manually waving them until the container is published, then we can merge, otherwise add an xfail and we remove it once the container is published. Pick your poison :upside_down_face:

**Comment by dirkmueller** [comment]
It feels like we need to merge this so that QA can approve the testing of rust1.66 and enable it? I think there is no harm in waiting

**Comment by jlausuch** [comment]
> It feels like we need to merge this so that QA can approve the testing of rust1.66 and enable it? I think there is no harm in waiting

Yes, openQA should not care if the image is not in public registry as it will pull it from IBS.

**Comment by dcermak** [comment]
> > It feels like we need to merge this so that QA can approve the testing of rust1.66 and enable it? I think there is no harm in waiting
> 
> Yes, openQA should not care if the image is not in public registry as it will pull it from IBS.

That's not the point of that metadata test. That test checks whether the reference is correct, that has nothing to do with where the image came from, but whether its canonical source is correct (which it can't be if the image has not been published yet).

**Pull request merged**